addons/crm/models/crm_lead.py

```diff
@@ -916,17 +916,19 @@ def _get_lead_duplicates(self, partner=None, email=None, include_lost=False):
           (still only active leads are considered). If False, search for active
           and not won leads and opportunities;
         """
-        if not email:
+        if not email and not partner:
             return self.env['crm.lead']
-        partner_match_domain = []
-        for email in set(email_split(email) + [email]):
-            partner_match_domain.append(('email_from', '=ilike', email))
+
+        domain = []
+        for normalized_email in [tools.email_normalize(email) for email in tools.email_split(email)]:
```

Comment:
Just a quite old thinking :
I'm wondering at what point we should make email_normalize = a string list of email normalized.
We could do for email in email_split(self.email_normalized) directly, everywhere.
But it could become super annoying to use.. for Now, I think we consider the email_normalize field contain only the first email of the email string list. (because email field can contain multiple email addresses).
There is something to do somehow here I think.. But maybe not today..

Reply:
why not keeping `for normalized_email in **set**(tools.email_normalize(email) for email in tools.email_split(email))` ?

Reply:
I adivse you to note that in a task so that we can have a chat about it afterwards ... this one begins to give me boutons.